### PR TITLE
modified teleport commands to work (mostly) like mineplex

### PR DESCRIPTION
**Pull request by ezrn**
- Can teleport to home from anywhere, with no cooldown
- /c home takes 20 seconds to teleport
- /c stuck takes 1 minute normally and 2 minutes while in enemy territory
- Can no longer start teleports while combat-tagged
- Teleport instant home from inside your claim(if there are no enemies) and from safezones
- In enemy territory c home will take 2 mins and in neutral territory it will take 1

### CODE REVIEW

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java [hunk 1]
```diff
@@ -109,27 +111,45 @@ public void displayOwner(Player player, Clan locationClan) {
...
                 if (clan.isSafe() && clan.getName().contains("Spawn") && event.getPlayer().getLocation().getY() > 110) {
                     return;
                 }
             }
 
-            UtilMessage.message(event.getPlayer(), "Clans", "You can only teleport to your clan home from spawn or the wilderness.");
-            event.setCancelled(true);
+            if (clanManager.getRelation(clanManager.getClanByPlayer(event.getPlayer()).orElse(null), clan) == ClanRelation.ENEMY) {
+                UtilMessage.message(event.getPlayer(), "Clans", "You cannot teleport to your clan home from enemy territory.");
+                event.setCancelled(true);
```
Comment (Mykindos) [inline clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java:131]: Maybe make this the same duration as c stuck (2 minutes)

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java [hunk 2]
```diff
@@ -146,33 +166,12 @@ public void onClanStuckTeleport(ClanStuckTeleportEvent event) {
...
-            if (clanManager.getClanByLocation(event.getPlayer().getLocation()).isPresent()) {
-                UtilMessage.message(event.getPlayer(), "Spawn", "You can only teleport to spawn from the wilderness.");
-                event.setCancelled(true);
-            } else {
-                event.setDelayInSeconds(30);
-            }
-
-        });
-
-    }
```
Comment (Mykindos) [inline clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java:177]: Honestly leave this in, the spawn command can just be disabled for players via config in-case we want to re-enable in the future

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java (hunk 1 above)
Comment (ezrn) [reply clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java:131]: Should I then make it 1 minute in any other territory aswell? And if we are allowing it In shops now it should probably be instant there given that you arent combat tagged.

Comment (ezrn) [reply clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java]: did this

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java [hunk 3]
```diff
@@ -119,21 +119,19 @@ public void onClanHomeTeleport(ClanHomeTeleportEvent event) {
...
             }
 
-            if (clanManager.getRelation(clanManager.getClanByPlayer(event.getPlayer()).orElse(null), clan) == ClanRelation.ENEMY) {
-                UtilMessage.message(event.getPlayer(), "Clans", "You cannot teleport to your clan home from enemy territory.");
-                event.setCancelled(true);
+            if (clanManager.getRelation(clanManager.getClanByPlayer(player).orElse(null), clan) == ClanRelation.ENEMY) {
+                event.setDelayInSeconds(120);
             } else {
-                event.setDelayInSeconds(30);
+                event.setDelayInSeconds(60);
```
Comment (Mykindos) [inline clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java:131]: should it be 60 seconds from your own claim

Comment (Mykindos) [reply clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClansMovementListener.java:131]: Should you even be allowed to c home from shops

**Review by Mykindos — APPROVED** [review]

**Pull request merged**